addons/web/static/src/legacy/js/core/menu.js

```diff
@@ -61,7 +61,10 @@ export async function initAutoMoreMenu(el, options) {
 
     const debouncedAdapt = () => {
         clearTimeout(debounce);
-        debounce = setTimeout(_adapt, 250);
+        debounce = setTimeout(() => {
+            el.dispatchEvent(new Event("autoMoreMenu.willAdapt", {bubbles: true}));
+            _adapt();
```

Comment:
Why is not the `autoMoreMenu.willAdapt` event triggered in the `_adapt` function instead of the debounced version only?

Reply:
Just because the issues are mainly caused by the debounced code (as described in the commit message). But Yes, I think it's better to have it in the `_adapt()` to prevent any possible issue caused by the unbreakable rollback... Updated now :+1: